kvstore/simplemap.go

```diff
@@ -0,0 +1,75 @@
+package kvstore
+
+// MapStore defines an interface that represents a key-value store that backs
+// the SM(S)T. It is a subset of the full functionality of a key-value store
+// needed for the SM(S)T to function. By using a simplified interface any
+// key-value store that implements these methods can be used with the SM(S)T.
+type MapStore interface {
+	// Accessors
+	Get(key []byte) ([]byte, error)
+	Set(key, value []byte) error
+	Delete(key []byte) error
+	Len() int
+
+	// Debug
+	ClearAll() error
+}
+
+// simpleMap is a simple in-memory map.
```

Comment:
1. Consider splitting the implementation & interface into separate files (see comment above)
2. Assert struct implements the interface (set patterns others will copy-paste)